dashboard/app/models/hint_view_request.rb

```diff
@@ -0,0 +1,43 @@
+# == Schema Information
+#
+# Table name: hint_view_requests
+#
+#  id            :integer          not null, primary key
+#  user_id       :integer
+#  script_id     :integer
+#  level_id      :integer
+#  feedback_type :integer
+#  feedback_xml  :text(65535)
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
```

Comment:
not null for both updated_at and created_at should be fine because when a record is created it sets both created_at and updated_at.

Did the not-null-ness of the timestamps just start appearing in the default migration generator in Rails 4.2 (along with the foreign keys)?

(you could verify that in a unit test that just checks that simply creating one of these does not crash ;))